st_preview/preview_math.py

```diff
@@ -143,15 +144,51 @@ def _create_image(latex_program, latex_document, base_name, color,
         ])
 
     err_file_path = image_path + _ERROR_EXTENSION
+    err_log = []
     if not pdf_exists:
-        with open(err_file_path, "w") as f:
-            f.write(
-                "Failed to run '{latex_program}' to create pdf to preview."
-                .format(**locals())
-            )
+        err_log.append(
+            "Failed to run '{latex_program}' to create pdf to preview."
+            .format(**locals())
+        )
+        err_log.append("")
+        err_log.append("")
+        err_log.append("LaTeX document:")
+        err_log.append("-----BEGIN DOCUMENT-----")
+        err_log.append(latex_document)
+        err_log.append("-----END DOCUMENT-----")
+
+        err_log.append("")
+
+        log_file = os.path.join(temp_path, base_name + ".log")
+        if not os.path.exists(log_file):
+            err_log.append("No log file found.")
+        else:
+            with open(log_file, "rb") as f:
+                log_data = f.read()
+            try:
+                errors, warnings, _ = parse_tex_log(log_data, temp_path)
+            except StopIteration:
+                err_log.append("Error while parsing log file.")
+                errors = warnings = []
+            if errors:
+                err_log.append("Logged errors:")
+                err_log.extend(errors)
+            if warnings:
+                err_log.append("Logged warnings:")
+                err_log.extend(warnings)
+            err_log.append("")
+
+            log_content = log_data.decode("utf8")
```

Comment:
Might be good to make this `log_data.decode('utf-8', 'ignore')`. Technically the log files are written using LaTeX's output-encoding. For LuaLaTeX and XeLaTeX this is always UTF-8, but for PdfLaTeX, its usually T1 which is *mostly* similar to `utf-8` (at least for the Latin alphabet) but sometimes differs.